src/adapter/src/catalog.rs

```diff
@@ -2036,9 +2078,11 @@ impl Source {
     pub fn user_controllable_persist_shard_count(&self) -> i64 {
         match &self.data_source {
             DataSourceDesc::Ingestion(ingestion) => {
-                // Ingestions with subsources only use persist shards for their subsources; those
-                // without subsources use 1.
-                std::cmp::max(1, i64::try_from(ingestion.subsource_exports.len()).expect("fewer than i64::MAX persist shards"))
+                // Ingestions with subsources only use persist shards for their
+                // subsources (i.e. not the primary source's persist shard);
```

Comment:
Is this true? I remember being able to query the main source and get back an empty result

Reply:
I suggested that it isn't "fair" to bill customers for this shard because it's of no use to them and other folks agreed. So I think the statement is not entirely true in a pedantic sense, but is true in the sense that "the persist  shard for the primary source is of no use."